src/core/iomgr/pollset_kick_posix.h

```diff
@@ -39,13 +39,49 @@
 
 typedef struct grpc_kick_fd_info {
   grpc_wakeup_fd_info wakeup_fd;
+  /* used for polling list and free list */
   struct grpc_kick_fd_info *next;
+  /* only used when polling */
+  struct grpc_kick_fd_info *prev;
 } grpc_kick_fd_info;
 
 typedef struct grpc_pollset_kick_state {
   gpr_mu mu;
   int kicked;
-  struct grpc_kick_fd_info *fd_info;
+  struct grpc_kick_fd_info fd_list;
 } grpc_pollset_kick_state;
 
-#endif  /* GRPC_INTERNAL_CORE_IOMGR_POLLSET_KICK_POSIX_H */
+#define GRPC_POLLSET_KICK_GET_FD(kick_fd_info) \
```

Comment:
The concept of pollset kicking isn't documented (fully) in a single place. Maybe add an overview at the beginning of the file and a few one-liner docstrings for these functions/macros?